include/interface/blas1_interface.h

```diff
@@ -491,55 +524,60 @@ typename sb_handle_t::event_t _dot(sb_handle_t &sb_handle, index_t _N,
  * @param _vy Buffer holding input vector y
  * @param _incy Stride of vector y (i.e. measured in elements of _vy)
  * @param _rs Output buffer
+ * @param _dependencies Vector of events
  * @return Vector of events to wait for.
  */
 template <typename sb_handle_t, typename container_0_t, typename container_1_t,
           typename container_2_t, typename index_t, typename increment_t>
-typename sb_handle_t::event_t _sdsdot(sb_handle_t &sb_handle, index_t _N,
-                                      float sb, container_0_t _vx,
-                                      increment_t _incx, container_1_t _vy,
-                                      increment_t _incy, container_2_t _rs) {
-  return internal::_sdsdot(sb_handle, _N, sb, _vx, _incx, _vy, _incy, _rs);
+typename sb_handle_t::event_t _sdsdot(
+    sb_handle_t &sb_handle, index_t _N, float sb, container_0_t _vx,
+    increment_t _incx, container_1_t _vy, increment_t _incy, container_2_t _rs,
+    typename sb_handle_t::event_t _dependencies = {}) {
+  return internal::_sdsdot(sb_handle, _N, sb, _vx, _incx, _vy, _incy, _rs,
+                           _dependencies);
 }
 
 /**
  * \brief ASUM Takes the sum of the absolute values
  * @param sb_handle SB_Handle
  * @param _vx BufferIterator
  * @param _incx Increment for the vector X
+ * @param _dependencies Vector of events
```

Comment:
@muhammad-tanvir-1211  thank you for your great work!
 I think it is missing the `rs` description in the documentation tag